src/document.h

```diff
@@ -2083,75 +2088,55 @@ struct Document {
 
     void PasteSingleText(Cell *c, const wxString &t) { c->text.Insert(this, t, selected); }
 
-    void PasteOrDrop() {
+    void PasteOrDrop(const wxTextDataObject &pdataobjt,
+                     const wxBitmapDataObject &pdataobji,
+                     const wxFileDataObject &pdataobjf) {
         Cell *c = selected.ThinExpand(this);
         if (!c) return;
         wxBusyCursor wait;
-        switch (sys->dataobjc->GetReceivedFormat().GetType()) {
-            case wxDF_FILENAME: {
-                const wxArrayString &as = sys->dataobjf->GetFilenames();
+
+        if (pdataobjt.GetText() != wxEmptyString) {

```

Comment:
maybe add some comments to the code explaining why we do things the way we do? why certain things are necessary on what platform? So we don't accidentally change it back in the future?

Reply:
you add a detail to the commit and the wxWidgets bug, but for long term knowledge, its best to be in the code.

Reply:
Hello Wouter
Thanks for your feedback. Please see the commit added. Would this be ok for you?
Thanks for your reply in advance!